### PR TITLE
faq/upgrade: update upgrade faq

### PR DESCRIPTION
**Pull request by crazycs520**
### What is changed, added or deleted? 
faq/upgrade: update upgrade faq.

add doc for `treat-old-version-utf8-as-utf8mb4`  config.

### What is the related PR or file link(s)? 

https://github.com/pingcap/tidb/pull/9820

### Which version does your change affect? 

dev, v3.0, v2.1, v2.1-legacy

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/faq/upgrade.md [hunk 1]
```diff
@@ -203,3 +203,7 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
         # Disabled.
         set @@session.tidb_check_mb4_value_in_utf8 = 0;
         ```
+
+- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是，把所有字符集为 UTF8 的表和 column 自动在转成 UTF8MB4 的 charset。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。所以在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4`，重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集任然还是 UTF8。
```
Comment (bb7133) [inline dev/faq/upgrade.md]:
```suggestion
- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是，把所有字符集为 UTF8 的表和 column 自动在转成 UTF8MB4 的 charset。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```

Comment (crazycs520) [reply dev/faq/upgrade.md]: done. thanks.

File: dev/faq/upgrade.md [hunk 2]
```diff
@@ -203,3 +203,7 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
         # Disabled.
         set @@session.tidb_check_mb4_value_in_utf8 = 0;
         ```
+
+- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是把 v2.1.7 版本之前创建的旧版本的表，自动把旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```
Comment (lilin90) [inline dev/faq/upgrade.md]:
```suggestion
- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开。该配置的作用是自动把 v2.1.7 版本之前创建的旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中将 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```

File: v2.1-legacy/op-guide/upgrade-faq.md [hunk 3]
```diff
@@ -202,4 +202,9 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
 
         # Disabled.
         set @@session.tidb_check_mb4_value_in_utf8 = 0;
-        ```
\ No newline at end of file
+        ```
+
+- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是把 v2.1.7 版本之前创建的旧版本的表，自动把旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```
Comment (lilin90) [inline v2.1-legacy/op-guide/upgrade-faq.md]:
```suggestion
- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开。该配置的作用是自动把 v2.1.7 版本之前创建的旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中将 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```

File: dev/faq/upgrade.md [hunk 4]
```diff
@@ -203,3 +203,7 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
...
+
+    注意：`treat-old-version-utf8-as-utf8mb4` 这个参数默认是打开的，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```
Comment (lilin90) [inline dev/faq/upgrade.md]:
```suggestion
    > **注意：**
    >
    > `treat-old-version-utf8-as-utf8mb4` 参数默认打开，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```

File: v2.1/faq/upgrade.md [hunk 5]
```diff
@@ -203,3 +203,7 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
         # Disabled.
         set @@session.tidb_check_mb4_value_in_utf8 = 0;
         ```
+
+- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是把 v2.1.7 版本之前创建的旧版本的表，自动把旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```
Comment (lilin90) [inline v2.1/faq/upgrade.md]:
```suggestion
- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开。该配置的作用是自动把 v2.1.7 版本之前创建的旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中将 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```

File: v3.0/faq/upgrade.md [hunk 6]
```diff
@@ -204,3 +204,8 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
         # Disabled.
         set @@session.tidb_check_mb4_value_in_utf8 = 0;
         ```
+
+- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格的要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开，这个配置的作用是把 v2.1.7 版本之前创建的旧版本的表，自动把旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中把 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```
Comment (lilin90) [inline v3.0/faq/upgrade.md]:
```suggestion
- v2.1.7 及之后版本，如果对表和 column 的字符集没有严格要求为 UTF8，也不想修改客户端代码去跳过 UTF8 检查或者手动修改 column 的 charset，可以在配置文件中把 `treat-old-version-utf8-as-utf8mb4` 打开。该配置的作用是自动把 v2.1.7 版本之前创建的旧版本的表和 column 的 UTF8 字符集转成 UTF8MB4。这个转换是在 TiDB load schema 时在内存中将 UTF8 转成 UTF8MB4，不会对实际存储的数据做任何修改。在配置文件中关闭 `treat-old-version-utf8-as-utf8mb4` 并重启 TiDB 后，以前字符集为 UTF8 的表和 column 的字符集仍然还是 UTF8。
```

File: v2.1-legacy/op-guide/upgrade-faq.md [hunk 7]
```diff
@@ -202,4 +202,9 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
...
+
+    注意：`treat-old-version-utf8-as-utf8mb4` 这个参数默认是打开的，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```
Comment (lilin90) [inline v2.1-legacy/op-guide/upgrade-faq.md]:
```suggestion
    > **注意：**
    >
    > `treat-old-version-utf8-as-utf8mb4` 参数默认打开，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```

File: v2.1/faq/upgrade.md [hunk 8]
```diff
@@ -203,3 +203,7 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
...
+
+    注意：`treat-old-version-utf8-as-utf8mb4` 这个参数默认是打开的，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```
Comment (lilin90) [inline v2.1/faq/upgrade.md]:
```suggestion
    > **注意：**
    >
    > `treat-old-version-utf8-as-utf8mb4` 参数默认打开，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```

File: v3.0/faq/upgrade.md [hunk 9]
```diff
@@ -204,3 +204,8 @@ TiDB 在 v2.1.1 及之前版本中，如果 charset 是 UTF8，没有对 4-byte
...
+
+    注意：`treat-old-version-utf8-as-utf8mb4` 这个参数默认是打开的，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```
Comment (lilin90) [inline v3.0/faq/upgrade.md]:
```suggestion
    > **注意：**
    >
    > `treat-old-version-utf8-as-utf8mb4` 参数默认打开，如果客户端强制需要用 UTF8 而不用 UTF8MB4，需要在配置文件中关闭。
```

**Pull request merged**